clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/IngestionThrottler.java

```diff
@@ -69,6 +74,37 @@ public IngestionThrottler(
         false,
         EventThrottler.BLOCK_STRATEGY);
 
+    this.aaWCLeaderRecordThrottler = new EventThrottler(
```

Comment:
Besides, wondering if it is possible that we could lazily instantiate the per-pool throttler inside this class when creating KCS, and then KCS could have `IntConsumer poolTypeThrottlerFunc` passing ingestionThrottler.maybeThrottleRecordRate(ConsumerPoolType poolType, int recordCount) to `ConsumptionTask`.

Reply:
Honestly, I don't think such optimization is required from perf POV and I would like to keep the code structure simple, such as keeping all the throttlers inside one class instead of spreading them in different classes.

Reply:
Oh, this is not for perf POV, but for code simplicity. I mean we could have a general `ingestionThrottler.maybeThrottleRecordRate(ConsumerPoolType poolType, int recordCount)` to trigger per-type throttler creation inside `IngestionThtrottler` and throttle action.  So KCS constructor does not need to check different type to create specific `IntConsumer poolTypeThrottlerFunc` function. All throttles are still inside `IngestionThtrottler`.

Reply:
Sure, I can make the change to move the mapping logic from KCS to `IngestionThrottler`.